embedded-c++/usearch/generate_ground_truth.cpp

```diff
@@ -3,245 +3,348 @@
 #include "usearch/helpers/index_operations.h"
 #include "usearch/helpers/query_runner.h"
 #include <iostream>
+#include <mutex>
+#include <thread>
+#include <vector>
+#include <cmath>
+
+
+
 
 using namespace duckdb;
 
+
+
+
 // ==================== Ground Truth Generator ====================
 class GroundTruthGenerator {
 private:
-	DuckDB db;
-	Connection con;
-	std::vector<DatasetConfig> datasets;
-	int numThreads;
-
-    // Structure to hold vector data
-    struct VectorData {
-        int id;
-        std::vector<float> vec;
-    };
-
-    // Structure to hold distance result
-    struct DistanceResult {
-        int query_id;
-        int neighbor_id;
-        float distance;
-    };
-
-    // Compute L2 (Euclidean) distance between two vectors
-    float computeDistance(const std::vector<float>& vec1, const std::vector<float>& vec2) {
-        float sum = 0.0f;
-        for (size_t i = 0; i < vec1.size(); i++) {
-            float diff = vec1[i] - vec2[i];
-            sum += diff * diff;
-        }
-        return std::sqrt(sum);
-    }
-
-    // Worker function for multi-threaded processing
-    void processTestVectorBatch(
-        const std::vector<VectorData>& testVectors,
-        const std::vector<VectorData>& trainVectors,
-        size_t start_idx,
-        size_t end_idx,
-        std::vector<DistanceResult>& results,
-        std::mutex& resultsMutex
-    ) {
-        std::vector<DistanceResult> localResults;
-        
-        // For each test vector in this batch
-        for (size_t i = start_idx; i < end_idx && i < testVectors.size(); i++) {
-            const auto& testVec = testVectors[i];
-            
-            // Calculate distance to all training vectors
-            for (const auto& trainVec : trainVectors) {
-                DistanceResult result;
-                result.query_id = testVec.id;
-                result.neighbor_id = trainVec.id;
-                result.distance = computeDistance(testVec.vec, trainVec.vec);
-                localResults.push_back(result);
-            }
-            
-            // // Progress indicator every 10 test vectors
-            // if ((i - start_idx) % 10 == 0) {
-            //     std::cout << "Thread processed " << (i - start_idx) << " test vectors" << std::endl;
-            // }
+  DuckDB db;
+  Connection con;
+  std::vector<DatasetConfig> datasets;
+  int numThreads;
+
+
+
+
+  // Structure to hold vector data
+  struct VectorData {
+      int id;
+      std::vector<float> vec;
+  };
+
+
+  struct DistanceResult {
+   int query_id;
+   int neighbor_id;
+   float distance;
+};
+
+
+
+
+  // Compute L2 (Euclidean) distance between two vectors
+  float computeDistance(const std::vector<float>& vec1, const std::vector<float>& vec2) {
+      float sum = 0.0f;
```

Comment:
Can you fix your IDE so there's not a bajillion lines between every code segment jfc😭💀